docker/Makefile

```diff
@@ -1,96 +1,53 @@
+HAIL_UBUNTU_IMAGE := $(DOCKER_PREFIX)/hail-ubuntu:$(TOKEN)
+BASE_IMAGE := $(DOCKER_PREFIX)/base:$(TOKEN)
+SERVICE_BASE_IMAGE := $(DOCKER_PREFIX)/service-base:$(TOKEN)
+SERVICE_JAVA_RUN_BASE_IMAGE := $(DOCKER_PREFIX)/service-java-run-base:$(TOKEN)
+HAIL_PUBLIC_IMAGE := $(DOCKER_PREFIX)/hail-public:$(TOKEN)
+GENETICS_PUBLIC_IMAGE := $(DOCKER_PREFIX)/genetics-public:$(TOKEN)
 
-SERVICE_BASE_LATEST = $(DOCKER_PREFIX)/service-base:latest
-SERVICE_BASE_IMAGE = $(DOCKER_PREFIX)/service-base:$(shell docker images -q --no-trunc service-base:latest | sed -e 's,[^:]*:,,')
-
-SERVICE_JAVA_RUN_BASE_LATEST = $(DOCKER_PREFIX)/service-java-run-base:latest
-SERVICE_JAVA_RUN_BASE_IMAGE = $(DOCKER_PREFIX)/service-java-run-base:$(shell docker images -q --no-trunc service-java-run-base:latest | sed -e 's,[^:]*:,,')
-
-HAIL_PUBLIC_LATEST = $(DOCKER_PREFIX)/hail-public:latest
-HAIL_PUBLIC_IMAGE = $(DOCKER_PREFIX)/hail-public:$(shell docker images -q --no-trunc hail-public:latest | sed -e 's,[^:]*:,,')
-
-GENETICS_PUBLIC_LATEST = $(DOCKER_PREFIX)/genetics-public:latest
-GENETICS_PUBLIC_IMAGE = $(DOCKER_PREFIX)/genetics-public:$(shell docker images -q --no-trunc genetics-public:latest | sed -e 's,[^:]*:,,')
-
-.PHONY: hail-ubuntu
 hail-ubuntu: hail-ubuntu-stmp
-
 hail-ubuntu-stmp: hail-ubuntu/Dockerfile hail-ubuntu/hail-pip-install hail-ubuntu/pip.conf hail-ubuntu/hail-apt-get-install
-	-docker pull $(DOCKER_PREFIX)/ubuntu:bionic-20200921
-	-docker pull $(HAIL_UBUNTU_LATEST)
 	python3 ../ci/jinja2_render.py '{"global":{"docker_prefix":"$(DOCKER_PREFIX)"}}' hail-ubuntu/Dockerfile hail-ubuntu/Dockerfile.out
-	docker build -t hail-ubuntu -f hail-ubuntu/Dockerfile.out --cache-from hail-ubuntu,$(HAIL_UBUNTU_LATEST),ubuntu:bionic-20200921 hail-ubuntu
-	touch hail-ubuntu-stmp
-
-.PHONY: base
-base: base-stmp
+	../docker-build.sh hail-ubuntu Dockerfile.out $(HAIL_UBUNTU_IMAGE)
+	echo $(HAIL_UBUNTU_IMAGE) > hail-ubuntu-stmp
 
-base-stmp: hail-ubuntu-stmp Dockerfile.base core-site.xml requirements.txt ../pylintrc ../setup.cfg
-	-docker pull $(BASE_LATEST)
-	python3 ../ci/jinja2_render.py '{"hail_ubuntu_image":{"image":"hail-ubuntu"}}' Dockerfile.base Dockerfile.base.out
-	[ "$(shell bash stat-permissions.sh Dockerfile.base)" = "644" ]
-	[ "$(shell bash stat-permissions.sh core-site.xml)" = "644" ]
-	[ "$(shell bash stat-permissions.sh requirements.txt)" = "644" ]
-	[ "$(shell bash stat-permissions.sh ../pylintrc)" = "644" ]
-	[ "$(shell bash stat-permissions.sh ../setup.cfg)" = "644" ]
-	docker build -t base -f Dockerfile.base.out --cache-from base,$(BASE_LATEST),hail-ubuntu ..
-	touch base-stmp
+base: hail-ubuntu-stmp Dockerfile.base core-site.xml requirements.txt ../pylintrc ../setup.cfg
+	python3 ../ci/jinja2_render.py '{"hail_ubuntu_image":{"image":"'$$(cat hail-ubuntu-stmp)'"}}' Dockerfile.base Dockerfile.base.out
+	../docker-build.sh .. docker/Dockerfile.base.out $(BASE_IMAGE)
+	echo $(BASE_IMAGE) > base
```

Comment:
The use of the `-stmp` convention seems inconsistent to me. Looks like it's only `hail-ubuntu-stmp` now? I don't care much whether to use it or not but would prefer whichever way we can be consistent.

Reply:
I switched everything to -image-ref. We can't use `hail-ubuntu` because that's the name of a directory containing the hail-ubuntu files.